### PR TITLE
Fixed some parameter to work with my Danfoss Air 

### PR DESCRIPTION
**Pull request by mschlappa**
Can you verify if it works for you too?

Greetings Marcus

### CODE REVIEW

**Comment by mschlappa** [comment]
Oh. I have to check "boost" command before merge. Stand by ;-)

**Comment by mschlappa** [comment]
Hi @JonasPed1 Now you can check the PR.

**Comment by mschlappa** [comment]
The last fix (2 commits) was a nasty one.  automatic_bypass read output was inverted and the command too :-)

**Comment by JonasPed** [comment]
Interesting that there, apparently, are different implementations. I will try to test it against my unit in a couple of days.

**Comment by mschlappa** [comment]
Thank you!

I found another small issue in the homeassistant Integration regarding the automatic_bypass switch. But I couldn’t test it yet.

It looks like there is an duplication of the bypass command where I would expect the automatic_bypass command

[=> diff](https://github.com/home-assistant/core/compare/dev...mschlappa:core:dev?diff=split&w=)

File: pydanfossair/danfossclient.py [hunk 1]
```diff
@@ -115,7 +119,7 @@ def _read_command(self, command, socket):
             return_value = self._read_bit(command, socket)
 
         if command == ReadCommand.automatic_bypass:
-            return_value = not self._read_bit(command, socket)
+            return_value = self._read_bit(command, socket)
```
Comment (JonasPed) [inline pydanfossair/danfossclient.py]: It looks like the response is inverted on my unit now.

File: pydanfossair/commands.py [hunk 2]
```diff
@@ -8,12 +8,12 @@ class ReadCommand(Enum):
...
     roomTemperatureCalculated = bytes([0x00, 0x04, 0x14, 0x96])
     humidity = bytes([0x01, 0x04, 0x14, 0x70])
-    bypass = bytes([0x00, 0x04, 0x14, 0x60])
+    bypass = bytes([0x01, 0x04, 0x14, 0x60])
     filterPercent = bytes([0x01, 0x04, 0x14, 0x6a])
     boost = bytes([0x01, 0x04, 0x15, 0x30])
     supply_fan_speed = bytes([0x00, 0x04, 0x14, 0x50])
     exhaust_fan_speed = bytes([0x00, 0x04, 0x14, 0x51])
-    fan_step = bytes([0x00, 0x04, 0x17, 0x81])
+    fan_step = bytes([0x01, 0x04, 0x15, 0x61])
```
Comment (JonasPed) [inline pydanfossair/commands.py:16]: The fan step is now a step between 1 and 10 and not the percentage of full speed (between 10 and 100).

File: pydanfossair/commands.py [hunk 3]
```diff
@@ -22,10 +22,10 @@ class ReadCommand(Enum):
...
     boost_activate = bytes([0x01, 0x06, 0x15, 0x30, 0x01])
     boost_deactivate = bytes([0x01, 0x06, 0x15, 0x30, 0x00])
-    bypass_activate = bytes([0x00, 0x06, 0x14, 0x60, 0x01])
-    bypass_deactivate = bytes([0x00, 0x06, 0x14, 0x60, 0x00])
-    automatic_bypass_deactivate = bytes([0x01, 0x06, 0x17, 0x06, 0x01])
-    automatic_bypass_activate = bytes([0x01, 0x06, 0x17, 0x06, 0x00])
+    bypass_activate = bytes([0x01, 0x06, 0x14, 0x60, 0x01])
+    bypass_deactivate = bytes([0x01, 0x06, 0x14, 0x60, 0x00])
+    automatic_bypass_deactivate = bytes([0x01, 0x06, 0x17, 0x06, 0x00])
+    automatic_bypass_activate = bytes([0x01, 0x06, 0x17, 0x06, 0x01])
```
Comment (JonasPed) [inline pydanfossair/commands.py]: I think automatic bypass is inverted now. Reading out all the data from the unit it says "disable automatic bypass", so I will assume that it must be set to true if it should be disabled.

**Review by JonasPed — CHANGES_REQUESTED** [review]
I have added a couple of comments. Please see below for the output before and after your changes. 

*Before changes* 
```
(venv) jonas@asterix-kubuntu:~/development/pydanfossair/pydanfoss-air (master)$ python -m pydanfossair --host <HOSTNAME>
ReadCommand.exhaustTemperature: 12.82
ReadCommand.supplyTemperature: 19.9
ReadCommand.extractTemperature: 20.62
ReadCommand.outdoorTemperature: 11.6
ReadCommand.roomTemperature: 21.01
ReadCommand.roomTemperatureCalculated: 20.29
ReadCommand.humidity: 54.11764705882353
ReadCommand.bypass: False
ReadCommand.filterPercent: 21.96078431372549
ReadCommand.boost: False
ReadCommand.supply_fan_speed: 1214
ReadCommand.exhaust_fan_speed: 1611
ReadCommand.fan_step: 40
ReadCommand.away_mode: False
ReadCommand.battery_percent: 0.0
ReadCommand.automatic_bypass: True
ReadCommand.operation_mode: program
```

*After changes*
```
(venv) jonas@asterix-kubuntu:~/development/pydanfossair/pydanfoss-air (mschlappa/master)$ python -m pydanfossair --host <HOSTNAME>
ReadCommand.exhaustTemperature: 12.92
ReadCommand.supplyTemperature: 19.97
ReadCommand.extractTemperature: 20.67
ReadCommand.outdoorTemperature: 11.7
ReadCommand.roomTemperature: 20.99
ReadCommand.roomTemperatureCalculated: 20.31
ReadCommand.humidity: 54.509803921568626
ReadCommand.bypass: False
ReadCommand.filterPercent: 21.96078431372549
ReadCommand.boost: False
ReadCommand.supply_fan_speed: 1242
ReadCommand.exhaust_fan_speed: 1612
ReadCommand.fan_step: 4
ReadCommand.away_mode: False
ReadCommand.battery_percent: 0.0
ReadCommand.automatic_bypass: False
ReadCommand.operation_mode: program
```

File: pydanfossair/danfossclient.py (hunk 1 above)
Comment (mschlappa) [reply pydanfossair/danfossclient.py]: You are right. I reverted the change.

File: pydanfossair/commands.py (hunk 3 above)
Comment (mschlappa) [reply pydanfossair/commands.py]: You are right. I reverted the change.

File: pydanfossair/danfossclient.py [hunk 4]
```diff
@@ -122,7 +126,7 @@ def _read_command(self, command, socket):
             return_value = self._read_short(command, socket)
 
         if command == ReadCommand.fan_step:
-            return_value = self._read_byte(command, socket)
+            return_value = self._read_byte(command, socket) * 10
```
Comment (mschlappa) [inline pydanfossair/danfossclient.py:129]: I multiplied the step with 10 so that it represents percentage again

**Comment by JonasPed** [comment]
Thanks for the PR.

**Comment by mschlappa** [comment]
Thank you for integrating my PR !!
Did you have a chance to look into this small change?
Maybe it can also integrated in the next Release of HA if you agree on it?

> 
> I found another small issue in the homeassistant Integration regarding the automatic_bypass switch. But I couldn’t test it yet.
> 
> It looks like there is an duplication of the bypass command where I would expect the automatic_bypass command
> 
> [=> diff](https://github.com/home-assistant/core/compare/dev...mschlappa:core:dev?diff=split&w=)

Greetings!
Marcus

**Pull request merged**